src/ZiraLink.Api.Application/Services/CustomerService.cs

```diff
@@ -157,61 +142,25 @@ public async Task UpdateProfileAsync(string userId, string name, string family,
                 throw new ArgumentNullException(nameof(family));
 
             var customer = await _dbContext.Customers.SingleOrDefaultAsync(x => x.ExternalId == userId, cancellationToken);
-            if (customer == null) throw new NotFoundException(nameof(Customer));
-
-            var client = await InitializeHttpClientAsync(cancellationToken);
-
+            if (customer == null)
+                throw new NotFoundException(nameof(Customer), new List<KeyValuePair<string, object>>() { new KeyValuePair<string, object>(nameof(Customer.ExternalId), userId) });
+             
             var jsonObject = new
             {
                 UserId = userId,
                 Name = name,
                 Family = family
             };
-            var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
-            var baseUri = _idsUri;
-            var uri = new Uri(baseUri, "User");
-            var response = await client.PatchAsync(uri.ToString(), content);
 
-            var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
-            var userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            var userCreationResult = await _identityService.UpdateUserAsync(jsonObject, cancellationToken);
```

Comment:
userCreationResult should be userUpdatingResult

Reply:
Done